src/runtime/runtime_rpi3.go

```diff
@@ -0,0 +1,140 @@
+// +build rpi3
+
+package runtime
+
+import (
+	"machine"
+	"runtime/volatile"
+	"unsafe"
+	"device/arm"
+)
+
+const tickMicros = int64(1)
+
+type timeUnit int64
+
+//go:export sleepticks sleepticks
+func sleepTicks(n timeUnit) {
+	WaitMuSec(uint32(n))
+}
+
+func ticks() timeUnit {
+	return timeUnit(SysTimer())
+}
+
+func preinit() {
+	//XXX because interp.Unsupported would be thrown if we tried to get this to happen at initializer time
+	machine.Aux = (*machine.AuxPeripheralsRegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x00215000))
+	machine.GPIO = (*machine.GPIORegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x00200000))
+	machine.SysTimer = (*machine.SysTimerRegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x3000))
+
+
+
+	// Initialize .bss: zero-initialized global variables.
+	//ptr := unsafe.Pointer(&_sbss)
+	//for ptr != unsafe.Pointer(&_ebss) {
+	//	*(*uint32)(ptr) = 0
+	//	ptr = unsafe.Pointer(uintptr(ptr) + 4)
+	//}
```

Comment:
This looks necessary to me, why is it commented out?

Reply:
Because I could not figure out how to get it to work. I was getting some undefined symbols so I need to investigate.  After this PR gets merged, I will fix.

Reply:
This really needs to be fixed, otherwise many programs won't work correctly (globals will not be zeroed if necessary). What is the error you're getting?

Reply:
I put this back in the code, although at the moment I don't have any code that sets ebss!=sbss...

Reply:
What do you mean? `_ebss` and `_sbss` are set by the linker, so what comes between is what the compiler/linker put there (for example, some globals used by the heap that are zero-initialized).